### DISCUSSION TITLE
How can i install HyperHDR under Libreelec?

### DISCUSSION THREAD

**Discussion by Johnny1982fd** [awawa-dev/HyperHDR#484] [Q&A] [answered]
Hello,

How can i install HyperHDR under Libreelec?

I want to change from HyperionNG to HyperHDR but i didnt found a installation guide?

Thanks in advance

**Comment by Baytro** [comment]
I think there is no official installation guide or anything, I did it with this script from the hyperion forum: https://hyperion-project.org/forum/index.php?thread/10463-install-hyperion-ng-on-libreelec-x86-64-rpi-inoffiziell-unofficially/. Then just change the name of the local tar file. If your are on Raspberry Pi 4 this should do it if you execute it at the location where the tar file is, change its name to whatever version you want to use:
tar -xzf "HyperHDR-18.0.0.0-Linux-armv7l.tar.gz" --strip-components=1 -C /storage share/hyperhdr/

chmod +x -R /storage/hyperhdr/bin

SERVICE_CONTENT="[Unit]
Description=HyperHDR ambient light systemd service
After=network.target

[Service]
Environment=DISPLAY=:0.0
ExecStart=/storage/hyperhdr/bin/hyperhdr --userdata /storage/hyperhdr/
TimeoutStopSec=2
Restart=always
RestartSec=10

[Install]
WantedBy=default.target"

systemctl -q enable hyperhdr.service --now

exit 0

These lines should unzip the tar file to /storage/hyperhdr and make the program executable (hyperhdr/bin). Then a service file is created so HyperHDR can be started and stopped with the following:
systemctl stop hyperhdr
systemctl start hyperhdr

**Reply by reedy-au** [reply]
Hey that's cool if it works.  

Do you use a rullz USB grabber?

ATM i'm using one intel nuc for librelec alpha builds(for HDR) and another nuc for hyperhdr.  I'm using a tesmart (https://www.amazon.com/dp/B07WT8SX1X?ref=ppx_yo2ov_dt_b_product_details&th=1) hdmi splitter which sometimes, well doesn't split when the second nuc talks to the grabber resulting in no TV picture.

I'll have to try this

**Reply by Johnny1982fd** [reply]
> I think there is no official installation guide or anything, I did it with this script from the hyperion forum: https://hyperion-project.org/forum/index.php?thread/10463-install-hyperion-ng-on-libreelec-x86-64-rpi-inoffiziell-unofficially/. Then just change the name of the local tar file. If your are on Raspberry Pi 4 this should do it if you execute it at the location where the tar file is, change its name to whatever version you want to use: tar -xzf "HyperHDR-18.0.0.0-Linux-armv7l.tar.gz" --strip-components=1 -C /storage share/hyperhdr/
> 
> chmod +x -R /storage/hyperhdr/bin
> 
> SERVICE_CONTENT="[Unit] Description=Hyperion ambient light systemd service After=network.target
> 
> [Service] Environment=DISPLAY=:0.0 ExecStart=/storage/hyperhdr/bin/hyperhdr --userdata /storage/hyperhdr/ TimeoutStopSec=2 Restart=always RestartSec=10
> 
> [Install] WantedBy=default.target"
> 
> systemctl -q enable hyperhdr.service --now
> 
> exit 0
> 
> These lines should unzip the tar file to /storage/hyperhdr and make the program executable (hyperhdr/bin). Then a service file is created so HyperHDR can be started and stopped with the following: systemctl stop hyperhdr systemctl start hyperhdr

Sounds difficult. ;(

**Reply by Baytro** [reply]
If you are familiar with Linux/Raspberry Pi it is not really difficult but it is definitely not just a click. I can help you If you want with a detailed step by step "Tutorial", it also took me a bit of time but was so worth it in the end ^^.

**Reply by Baytro** [reply]
> Hey that's cool if it works.
> 
> Do you use a rullz USB grabber?
> 
> ATM i'm using one intel nuc for librelec alpha builds(for HDR) and another nuc for hyperhdr. I'm using a tesmart (https://www.amazon.com/dp/B07WT8SX1X?ref=ppx_yo2ov_dt_b_product_details&th=1) hdmi splitter which sometimes, well doesn't split when the second nuc talks to the grabber resulting in no TV picture.
> 
> I'll have to try this

Im using a Raspberry Pi 4 with Libreelec and use it almost daily as my media center for Hi-Res music and movies. The ambilight works perfectly and I'm using a Ezcoo HDMI matrix so I can use ambilight for all my HDMI sources (FireTV, PS5, Raspberry Pi and Chromecast). I can control the Raspberry Pi and thus the ambilight with a remote (also another FireTV remote) that turns it on/off or switches HDR on a button press by executing a simple Script. This is the absolute best way for me and it's almost like it's built in officially.
As the USB Grabber I use just one that has the Macrosilican MS2109 chipset and thus the LUT table that was uploaded specifially for this grabber so HDR looks perfect. If you have any questions I hope I can help because I absolutely love this project and put a lot of time in it.

**Reply by Johnny1982fd** [reply]
> If you are familiar with Linux/Raspberry Pi it is not really difficult but it is definitely not just a click. I can help you If you want with a detailed step by step "Tutorial", it also took me a bit of time but was so worth it in the end ^^.

I will really appreciated it! I guess, more people will see this this thread/tutorial. Thanks in advance!!!

**Reply by Baytro** [reply]
Yeah probably, which platform are you on? Do you know how to Connect via ssh (PuTTY/WinSCP)?

**Reply by Johnny1982fd** [reply]
Im on a Raspberry PI 4 with Libreelec 10.0.3 (Kodi 19.4).
Im using Ambilight only with Kodi, SK6812 RGBW and WLED 0.3.13

Yes, i know how to connect via ssh. Im using terminal with my Mac.

**Reply by Baytro** [reply]
> Im on a Raspberry PI 4 with Libreelec 10.0.3 (Kodi 19.4). Im using Ambilight only with Kodi, SK6812 RGBW and WLED 0.3.13
> 
> Yes, i know how to connect via ssh. Im using terminal with my Mac.

Ok try this and let me know if it worked:

- log in via ssh
- `wget https://github.com/awawa-dev/HyperHDR/releases/download/v18.0.0.0/HyperHDR-18.0.0.0-Linux-armv7l.tar.gz` (downloads latest stable release version 18 of HyperHDR)
- `tar -xzf "HyperHDR-18.0.0.0-Linux-armv7l.tar.gz" --strip-components=1 -C /storage share/hyperhdr/` (extracts the file, a folder named "hyperhdr" should exist now)
- `chmod +x -R /storage/hyperhdr/bin` (gives the rights to make it executable)
- paste the following (creates a temporary variable named "SERVICE_CONTENT")
```
SERVICE_CONTENT="[Unit]
Description=HyperHDR ambient light systemd service
After=network.target

[Service]
Environment=DISPLAY=:0.0
ExecStart=/storage/hyperhdr/bin/hyperhdr --userdata /storage/hyperhdr/
TimeoutStopSec=2
Restart=always
RestartSec=10

[Install]
WantedBy=default.target"
```
- `echo "$SERVICE_CONTENT" > /storage/.config/system.d/hyperhdr.service`  (write content of variable into a service file)
- `systemctl -q enable hyperhdr.service --now` (enables the service)
- HyperHDR should be installed now and should also be active on reboot as it is a service. You can check the status with the following line: `systemctl status hyperhdr`. If it is active the service is running and everything works fine.

**Reply by awawa-dev** [reply]
For Gods sake please stop using Hyperion here ;) Of course you can put info about the source of the origin service/solution but yours modified code doesn't use Hyperion anymore.

**Reply by Baytro** [reply]
> For Gods sake please stop using Hyperion here ;) Of course you can put info about the source of the origin service/solution but yours modified code doesn't use Hyperion anymore.

I just used the reference from there and what to change so he can install HyperHDR instead of using Hyperion so I don't really know what you mean.

**Reply by Johnny1982fd** [reply]
> > For Gods sake please stop using Hyperion here ;) Of course you can put info about the source of the origin service/solution but yours modified code doesn't use Hyperion anymore.
> 
> I just used the reference from there and what to change so he can install HyperHDR instead of using Hyperion so I don't really know what you mean.

+1 , ill give a try, thank you so much

**Reply by Johnny1982fd** [reply]
Ill repsonse in the next days, pls dont close this.

**Reply by awawa-dev** [reply]
> > For Gods sake please stop using Hyperion here ;) Of course you can put info about the source of the origin service/solution but yours modified code doesn't use Hyperion anymore.
> 
> I just used the reference from there and what to change so he can install HyperHDR instead of using Hyperion so I don't really know what you mean.

that part:
![obraz](https://user-images.githubusercontent.com/69086569/211851863-9f1aeae7-2d9d-461d-a85c-6c3e760bcfad.png)

**Reply by Baytro** [reply]
Oh well sorry never looked into it you're right I just copy pasted it, will change it.

**Reply by awawa-dev** [reply]
OK, thanks 👍

**Reply by Johnny1982fd** [reply]
● hyperhdr.service - HyperHDR ambient light systemd service
     Loaded: loaded (/storage/.config/system.d/hyperhdr.service; enabled; vendor preset: disabled)
     Active: active (running) since Tue 2021-02-02 16:29:49 CET; 1 years 11 months ago
   Main PID: 589 (hyperhdr)
     CGroup: /system.slice/hyperhdr.service
             └─589 /storage/hyperhdr/bin/hyperhdr --userdata /storage/hyperhdr/

Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.779 SMOOTHING0             : <INFO> Clearing queued colors before: disabling
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.779 SMOOTHING0             : <INFO> Smoothing queue is cleared
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.780 LEDDEVICE_FILE         : <DEBUG> LedDevice.cpp:136:disable() | Disable device
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.780 LEDDEVICE_FILE         : <DEBUG> LedDevice.cpp:336:switchOff() | Switch off
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.780 LEDDEVICE_FILE         : <DEBUG> LedDevice.cpp:372:powerOff() | Power Off
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.780 LEDDEVICE_FILE         : <DEBUG> LedDevice.cpp:294:writeBlack() | Set LED strip to bla
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.780 LEDDEVICE_FILE         : <DEBUG> LedDeviceFile.cpp:83:close() | File: /dev/null
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.781 COMPONENTREG0          : <INFO> Video capture device: disabled
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.781 COMPONENTREG0          : <INFO> LED device: disabled
Jan 12 12:17:15 LibreELEC hyperhdr[589]: 2023-01-12T12:17:15.784 V4L2:AUTO              : <INFO> Stopped

~

**Reply by Baytro** [reply]
Nice so it worked? Looks good to me the service is active and running?

**Reply by Johnny1982fd** [reply]
Yes, it is running. I didnt configure HyperHDR yet, but i can invoke the Dashboard adress :)

So, is it possible to pin that for other users?

**Reply by Johnny1982fd** [reply]
In addition to that ... how can i update HyperHDR? Is it still to change the Download URL?

**Reply by Baytro** [reply]
Perfect then it worked ^^ I don't know if comments can be pinned but that would maybe make sense as more people are probably struggling with it. 
To change the version you only have to change the url and the name of the tar file and make it executable again with the same command. So probably only the version number. The rest should stay the same I think.

**Reply by Johnny1982fd** [reply]
I have been testing a little bit and im disappointed.

Im using HDRHDR 19 beta 19 with WLED 0.3.13. I have SK6812 RGB-NW 

Two things that i dont understand:

1.) Why are LEDs in black screens always on with white LEDs ???? (see attach)
![2023-01-15 17 09 55](https://user-images.githubusercontent.com/55974883/212553121-81e33cfa-2af9-4db2-90a8-5626ef4f08df.jpg)

2.) Where can i conigure white channel in HyperHDR?


3.) Why is my white from VideoCapture much different than in WLED?

**Reply by awawa-dev** [reply]
Probably you have set minimal brightness in the processing or your  "black" is not a real black. Black where the LED strip is powered down is RGB(0,0,0) everything else like RGB(1,1,1) will set LED strip to power on. And grabbers usually rarely return real black color. It can be adjusted but it requires from you to spent more time adjusting the processing settings.

**Reply by Johnny1982fd** [reply]
I cant confirm that.
It is the same if you take a black picture. It seems like it is a permanent white on, didnt know where it comes from.

I have a second installation with Hyperion NG, there is nothing like this.

**Reply by Johnny1982fd** [reply]
@awawa-dev 

i have problemlos to calibrate white. I have SK6812 RGBW. If i setup my whitechannel via WLED, these settings will be always overwrite by HyperHDR. (only calibrate RGB - not the white channel) 
How can i calibrate the whitechannel in HyperHDR?

**Reply by awawa-dev** [reply]
Only the brightness may be overwritten (can be disabled) and the live mode is set. Unfortunately I don't use WLED so can't help with its configuration.

**Reply by awawa-dev** [reply]
@Johnny1982fd  https://github.com/awawa-dev/HyperHDR/pull/353

**Comment by reedy-au** [comment]
This was quite helpful consolidating librelec and HyperHDR on the same NUC.  I put together a shell script to make life easier for others to install from the information here. It works on x86_64 haven't tested on RPI as I don't own one.

https://github.com/ghr12/hyperhdr

**Comment by awawa-dev** [comment, marked as answer]
Hi
Currently we have official native LibreELEC 11 addon. Read more here: https://github.com/awawa-dev/HyperHDR/discussions/600
There is no need to manually adjust system files and create custom scripts or services: just install the HyperHDR Kodi plug-in.
